addons/fleet/models/fleet_vehicle.py

```diff
@@ -202,10 +202,16 @@ def _compute_contract_reminder(self):
             total = 0
             name = ''
             state = ''
-            for element in record.log_contracts:
-                if element.state in ('open', 'expired') and element.expiration_date:
+
+             # Get the last contract by ordering by creation date
+            log_contract = self.env['fleet.vehicle.log.contract'].search([
+                ('vehicle_id', '=', record.id)
+                ], limit=1, order='create_date desc')
+
+            if log_contract and log_contract.expiration_date:
+                if log_contract.state in ('open', 'expired'):
                     current_date_str = fields.Date.context_today(record)
-                    due_time_str = element.expiration_date
+                    due_time_str = log_contract.expiration_date
```

Comment:
I think we also have to take into account 'new' contracts (but this needs to be verified with Geoffrey) as a contract that will start in a week will not yet be open or expired but should block the warning from appearing. 

Is the 'total' variable still relevant? It seems that with this change it will always be just 1 when a contract is overdue or due_soon and 0 in all other cases.

It also seems to me that instead of on creation date we should sort on expiration date, no? But this is also to be verified with Geoffrey.

Below this block of changed code, there new seems to be some code duplication where you determine log_contract again and do some more if-checks that are now irrelevant due to these changes.